### ISSUE TITLE
Performance tweaks relating to tools

### ISSUE THREAD

**Issue by klauer** [pcdshub/typhos#406]
* IPython Console should lazily load
   - IPython is a pretty heavy dependency, and (I believe) not so commonly used in the typhos suite
   - Lazily load IPython console only if the user clicks on it for the first time
* Plotting similarly should probably be refactored to load only when requested
* Others?

**Comment by ZLLentz** [comment]
Should we be forward-thinking and generically make all tools lazily load?
Though in my test run a few moments ago these tools seemed to load very quickly:
```
$ typhos xcs_ccm
[2020-10-02 09:57:07] - INFO - Loading Tools ...
[2020-10-02 09:57:08] - INFO - Adding devices ...
[2020-10-02 09:57:09] - INFO - Launching application ...
```

**Comment by klauer** [comment]
You're right, @ZLLentz - I think the IPython import is only around a half second from what I had measured with happi: not a huge deal overall.

Overall lazy-loading of tools may have a noticeable performance improvement 👍
